src/userbase-server/app.js

```diff
@@ -526,6 +575,78 @@ exports.listUsersWithPagination = async function (req, res) {
   }
 }
 
+exports.listUsersForDatabaseWithPagination = async function (req, res) {
+  let logChildObject
+  try {
+    const databaseId = req.params.databaseId
+    const nextPageToken = req.query.nextPageToken
+
+    logChildObject = { ...res.locals.logChildObject, databaseId, nextPageToken }
+    logger.child(logChildObject).info('Listing users from Admin API')
+
+    _validateDatabaseId(databaseId)
+
+    // get database stuff
+    const db = await dbController.findDatabaseByDatabaseId(databaseId)
+    if (!db) throw {
+      status: statusCodes['Not Found'],
+      error: { message: 'Database not found.' }
+    }
+    const ownerId = db['owner-id']
+
+    // get user stuff from db owner
+    const owner = await userController.getUserByUserId(ownerId)
+    if (!owner || owner.deleted) throw {
+      status: statusCodes['Not Found'],
+      error: { message: 'Database not found.' }
+    }
+    const appId = owner['app-id']
+
+    const lastEvaluatedKey = nextPageTokenToLastEvaluatedKey(
+      nextPageToken,
+      (lastEvaluatedKey) => _validateListUsersLastEvaluatedKey(lastEvaluatedKey, appId)
```

Comment:
Missed this on first pass -- I don't believe this is the correct validation for the key. I'm pretty sure it should be an object with key `'database-id'` and `'user-id'`

Reply:
Like this? (See db.js)

Reply:
Looks good! Will add tests to make sure it's working as expected.